src/PerformanceMonitor/PerformanceMonitor.cs

```diff
@@ -0,0 +1,245 @@
+using Akka.Actor;
+using Neo.ConsoleService;
+using Neo.IO.Json;
+using Neo.Ledger;
+using Neo.Network.P2P;
+using Neo.Network.P2P.Payloads;
+using Neo.Persistence;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Neo.Plugins
+{
+    public partial class PerformanceMonitor : Plugin, IPersistencePlugin, IP2PPlugin
+    {
+        public override string Name => "PerformanceMonitor";
+
+        public PerformanceMonitor()
+        {
+            RpcServerPlugin.RegisterMethods(this);
+        }
+
+        protected override void Configure()
+        {
+            Settings.Load(GetConfiguration());
+        }
+
+        private delegate void CommitHandler(StoreView snapshot);
+        private event CommitHandler OnCommitEvent;
+
+        public void OnCommit(StoreView snapshot)
+        {
+            OnCommitEvent?.Invoke(snapshot);
+        }
+
+        private delegate void PersistHandler(StoreView snapshot, IReadOnlyList<Blockchain.ApplicationExecuted> applicationExecutedList);
+        private event PersistHandler OnPersistEvent;
+
+        public void OnPersist(StoreView snapshot, IReadOnlyList<Blockchain.ApplicationExecuted> applicationExecutedList)
+        {
+            OnPersistEvent?.Invoke(snapshot, applicationExecutedList);
+        }
+
+        private delegate void P2PMessageHandler(Message message);
+        private event P2PMessageHandler OnP2PMessageEvent;
+
+        public bool OnP2PMessage(Message message)
+        {
+            OnP2PMessageEvent?.Invoke(message);
+            return true;
+        }
+
+        private delegate void ConsensusMessageHandler(ConsensusPayload payload);
+        private event ConsensusMessageHandler OnConsensusMessageEvent;
+
+        public bool OnConsensusMessage(ConsensusPayload payload)
+        {
+            OnConsensusMessageEvent?.Invoke(payload);
+            return true;
+        }
+
+        /// <summary>
+        /// Process "check cpu" command
+        /// Prints each thread CPU usage information every second
+        /// </summary>
+        [ConsoleCommand("check cpu", Category = "Check Commands", Description = "Show CPU usage information of each thread every second.")]
+        private void OnCheckCPUCommand()
+        {
```

Comment:
I think that we can merge cpu, threads, and memory in only one, it will be easier for check the current state.